firebase-ml-modeldownloader/src/test/java/com/google/firebase/ml/modeldownloader/FirebaseModelDownloaderTest.java

```diff
@@ -54,10 +91,32 @@ public void getModel_unimplemented() {
   }
 
   @Test
-  public void listDownloadedModels_unimplemented() {
-    assertThrows(
-        UnsupportedOperationException.class,
-        () -> FirebaseModelDownloader.getInstance().listDownloadedModels());
+  public void listDownloadedModels_returnsEmptyModelList()
+      throws ExecutionException, InterruptedException {
+    when(mockPrefs.listDownloadedModels()).thenReturn(Collections.emptySet());
+    TestOnCompleteListener<Set<CustomModel>> onCompleteListener = new TestOnCompleteListener<>();
+    Task<Set<CustomModel>> task = firebaseModelDownloader.listDownloadedModels();
+    task.addOnCompleteListener(executor, onCompleteListener);
+    Set<CustomModel> customModelSet = onCompleteListener.await();
```

Comment:
I wonder if `Tasks.await(task)` could be used instead here?

Reply:
I had tried that got - "java.lang.IllegalStateException: Must not be called on the main application thread", found this solution was used elsewhere.